Pantalla login

vivianachoque opened the pull request:
## Summary
Maqueté pantalla login.
Hice pantalla login con manejo de errores tanto en usuario, como en código de ingreso. Ambos muestran error en caso de estar incompleto 1 o ambos. 
Además uní login con API.
## Screenshots
![image](https://user-images.githubusercontent.com/56049159/78591488-e0d82a80-7819-11ea-826f-1fa2ebe6b738.png)
![image](https://user-images.githubusercontent.com/56049159/78591574-049b7080-781a-11ea-9e72-5df2fd5d6388.png)
![image](https://user-images.githubusercontent.com/56049159/78591613-13822300-781a-11ea-8d56-4f2c3dcf14a2.png)
![image](https://user-images.githubusercontent.com/56049159/78591671-31e81e80-781a-11ea-892e-4c6057d5156d.png)

## Invision
https://projects.invisionapp.com/d/#/console/19366632/404500026/preview
https://projects.invisionapp.com/d/#/console/19366632/404500027/preview
https://projects.invisionapp.com/d/#/console/19366632/404500028/preview
## Trello
https://trello.com/c/EKO6xGS5/2-maquetar-login
https://trello.com/c/BHjxRidh/6-manejo-de-errores-login
https://trello.com/c/ELThsc50/4-login-contra-backend

vivianachoque committed:
ordene las carpetas de componentes y empece a exportar los assets

vivianachoque committed:
cree componente blockimg, que son las imagenes de bloques en el login

vivianachoque committed:
exporte bloque

vivianachoque committed:
ordené carpetas

vivianachoque committed:
corregí ruteos

vivianachoque committed:
corregi rutas y carpetas

vivianachoque committed:
modifique rutas y córregí import de violetbutton

Ezequiiel98 committed:
acomadamos archivos y funciones

vivianachoque committed:
hice formulario de Login

vivianachoque committed:
hice carpeta para fonts

vivianachoque committed:
agregué fuente montserrat

vivianachoque committed:
corregi algunos error en codigo con min y mayus, también estoy estilizando componente Blockimg, ancho y alto

vivianachoque committed:
arreglé problema con google fonts, junto a tincho

vivianachoque committed:
uní componente Login con componente VioletB

vivianachoque committed:
agregue divs entre los labels, y separece button con divs

vivianachoque committed:
mejore algunas cosas del componente form

vivianachoque committed:
Alineé la columna de FormLogin, mejoré en gral. todos los márgenes de la pág

vivianachoque committed:
agregué style al form y al titulo

vivianachoque committed:
hice comments para corregir errores a futuro

vivianachoque committed:
modifiqué ancho de botones ya que estaba distinto a invision. Mejore padding de formLogin

vivianachoque committed:
modificando scroll no deseado de pantalla

vivianachoque committed:
mejore ubicación de img y saqué SCROLL no deseado

vivianachoque committed:
corregí margin entre form y blockimg

vivianachoque committed:
agregué distintos tipos de inputs al form, continuaré con issue

vivianachoque committed:
icon login error hice

vivianachoque committed:
corregi el @import de las clases para la clase .error

vivianachoque committed:
faltaba } en css de formLogin

vivianachoque committed:
cambie clase blockchain por title

vivianachoque committed:
corregí espacios en clases de formlogin

vivianachoque committed:
agregue img de error a login

vivianachoque committed:
corregí errores de anidación en css login form

vivianachoque committed:
hice cambios en .error

vivianachoque committed:
cree state para formLogn (error)

vivianachoque committed:
arregle Hover de WhiteButton

vivianachoque committed:
agregue state.error a input de NOMBRE

vivianachoque committed:
agregando states a FormLogin para manejo de errores login, corregí algunos errores de código

vivianachoque committed:
Agregué función a VioletButton (Disabled) y modifiqué algunas cosas de su clase

vivianachoque committed:
agregué type=submit al componente VioletButton

vivianachoque committed:
agregué state.errorName state.errorCode en caso de que ambas estén vacias va a haber ERROR en el form

vivianachoque committed:
uni api con login, hice manejo de errores de form

vivianachoque committed:
hice ruteo de api, modifique ruta

vivianachoque committed:
hice manejo de errores

vivianachoque committed:
le di styles a mensajes de error en login y agregué margin-top a containerButton

vivianachoque committed:
agregué archivos a git ignore link API

vivianachoque committed:
Corregí funciones de login error, ya que tenía algunos problemas a la hora de notificar algunos errores. Borré comentarioss innecesarios e hice nuevos para ir guiando a cualquier persona que vea mi código.

vivianachoque committed:
agregué una última línea vacía a cada archivo del proyecto

vivianachoque committed:
edité readme

vivianachoque committed:
agregué una linea a VioletButton (styles)

vivianachoque committed:
cambié ruta de services por Services y nombre de carpeta

MMarussi commented on .env:
```diff
@@ -0,0 +1 @@
+REACT_APP_API_URL=http://woloxblockchain.herokuapp.com
```
Cuidado con subir los .env a cualquier repositorio remoto ya que podría contener keys, tokens o configuraciones que son información sensible que no nos podemos dar el lujo que alguien sin los permisos adecuados vea.

Ahora sólo tiene la URL de un endpoint (que podría ser sensible dependiendo del endpoint), así que te sugiero borrar el archivo del repo remoto. Lo raro es que está especificado en el .gitignore que no se pueda agregar un ".env" a los commits :thinking:

vivianachoque replied:
Gracias :(, fue un error que notificamos, nos sirve de aprendizaje

MMarussi commented on .gitignore:
```diff
@@ -18,6 +18,9 @@
 yarn-error.log*
+
+# env 
+.env
```
Agrega una línea vacía al final porfavor

MMarussi commented on src/components/VioletButton/index.module.scss:
```diff
@@ -0,0 +1,47 @@
+  @extend .btn;
+  background-color: #7758A2;
+  color: white;
+  &:hover {
```
Agrega una línea vacía acá para separar el pseudo-selector de los atributos de la clase

vivianachoque replied:
Hecho

MMarussi commented on src/components/VioletButton/index.module.scss:
```diff
@@ -0,0 +1,47 @@
+  @extend .btn;
+  color: #B5B5B6;
+  background-color: #E8E8E8;
+}
```
No te olvides de la línea vacía al final

vivianachoque replied:
gracias

MMarussi commented on src/components/VioletButton/index.js:
```diff
@@ -0,0 +1,42 @@
+        <button
+          type="button"
+          className={disable ? styles.btnDisabled : styles.btnActive}
+          disable={disable.toString()}
```
```suggestion
          disabled={disable.toString()}
```

MMarussi commented on src/screens/LoginScreen/components/Blockimg/index.js:
```diff
@@ -0,0 +1,14 @@
+
+import styles from './index.module.scss';
+
+function Blockimg() {
```
Acordate que si el nombre de algo esta compuesto por 2 palabras o más se escribe en camelCase para que sea más fácil leerlo
```suggestion
function BlockImg() {
```

vivianachoque replied:
Listo gracias

MMarussi commented on src/screens/LoginScreen/components/Blockimg/index.js:
```diff
@@ -0,0 +1,14 @@
+
+function Blockimg() {
+  return (
+    <div className={styles.blockimg}>
```
```suggestion
    <div className={styles.blockImg}>
```

vivianachoque replied:
Gracias

MMarussi commented on src/screens/LoginScreen/components/Blockimg/index.module.scss:
```diff
@@ -0,0 +1,8 @@
+.blockimg {
```
```suggestion
.blockImg {
```

vivianachoque replied:
Gracias

MMarussi commented on src/screens/LoginScreen/components/FormLogin/index.js:
```diff
@@ -0,0 +1,92 @@
+
+  handleSendGame = async data => {
+    const res = await sendGame(data);
+    if (res.status === 500) {
+      this.setState({ errorCode: true });
+    } else if (res.status === 404) {
+      this.setState({ errorServer: true });
+    } else {
+      console.log('redireccionamiento');
+    }
```
Consulta: Hay más códigos de error? Por qué si no los hay es más cómodo reescribir el if para que la condición a chequear es que `res.status === 200`, así podes agrupar todos los errores en el else.

MMarussi commented on src/screens/LoginScreen/components/FormLogin/index.js:
```diff
@@ -0,0 +1,92 @@
+    } else {
+      const data = { name, code };
+      this.handleSendGame(data);
+    }
```
Fijate que no es necesario hacer el primer chequeo ya que `name` y `code` no son excluyentes entre si y se pueden validar por separado (es más, es necesario validarlos por separado).
Abajo sugerí una validación para realizar el send, pero hay que tener en cuenta que nada más contempla que el caso de error en el que `name` o `code` sean vacíos.
```suggestion
    if (name === '') {
      this.setState({ errorName: true });
    };
    if (code === '') {
      this.setState({ errorCode: true });
    };
    if (name && code) {
      const data = { name, code };
      this.handleSendGame(data);
    };
```

MMarussi commented on src/screens/LoginScreen/components/FormLogin/index.js:
```diff
@@ -0,0 +1,92 @@
+
+  render() {
+    return (
+      <form onSubmit={e => this.handleSubmit(e)} className={styles.formLogin}>
```
```suggestion
      <form onSubmit={this.handleSubmit} className={styles.formLogin}>
```
No es necesario especificar que recibe un evento como parametro ya que es un parámetro implícito.

MMarussi commented on src/screens/LoginScreen/components/FormLogin/index.js:
```diff
@@ -0,0 +1,92 @@
+          {this.state.errorName && <img src={img} className={styles.errorIcon} />}
+          <label className={styles.label}>NOMBRE</label>
+          <input
+            onChange={e => this.handleChange(e)}
```
Mismo que `onSubmit`

MMarussi commented on src/screens/LoginScreen/components/FormLogin/index.js:
```diff
@@ -0,0 +1,92 @@
+          {this.state.errorCode && <img src={img} className={styles.errorIcon} />}
+          <label className={styles.label}>CÓDIGO DE INGRESO</label>
+          <input
+            onChange={e => this.handleChange(e)}
```
Mismo que `onSubmit`

MMarussi commented on src/screens/LoginScreen/components/FormLogin/index.module.scss:
```diff
@@ -0,0 +1,124 @@
+    padding-top:50px;
+    flex-direction: column;
+    align-self: flex-start;
+
```
Está línea vacía se puede borrar

MMarussi commented on src/screens/LoginScreen/components/FormLogin/index.module.scss:
```diff
@@ -0,0 +1,124 @@
+    flex-direction: column;
+    padding: 8.5px 9.5px;
+    border-radius: 2px;
+    margin: 10px 5px 5px 0;
+    font-family: Montserrat;
+    font-size: 12px; 
+    border: 1px solid #BBBBBB;
+    outline-color: #EA4453;
+}
+
+.input:disabled {
+    flex-direction: column;
+    padding: 8.5px 9.5px;
+    border-radius: 2px;
+    margin: 10px 5px 5px 0;
+    font-family: Montserrat;
+    font-size: 12px;
+    outline-color: #B5B5B6;
+    background-color: #E8E8E8;
+}
+
+.input:hover {
+    flex-direction: column;
+    padding: 8.5px 9.5px;
+    border-radius: 2px;
+    margin: 10px 5px 5px 0;
+    font-family: Montserrat;
+    font-size: 12px;
+    outline-color: #525256;
+}
+
+.input:focus {
+    flex-direction: column;
+    padding: 8.5px 9.5px;
+    border-radius: 2px;
+    margin: 10px 5px 5px 0;
+    font-family: Montserrat;
+    font-size: 12px;
+    outline-color: #7758A2;
+}
```
Acá podés usar la sintaxis de SASS como hiciste en `src/components/VioletButton/index.module.scss`:

```SCSS
.input {
  flex-direction: column;
  padding: 8.5px 9.5px;
  border-radius: 2px;
  ...

  &:disabled {
    ...
  }

  &:hover {
    ...
  }

  ...
}
```

vivianachoque replied:
Hecho

MMarussi commented on src/screens/LoginScreen/components/FormLogin/index.module.scss:
```diff
@@ -0,0 +1,124 @@
+    outline-color: #7758A2;
+}
+
+ .error{
+    @extend .input;
+}
+
+ .error:focus{
+     outline-color: #EA4453;
+     border-color: #EA4453;
+ }
```
Mismo que `.input`

vivianachoque replied:
listo, quedó hermoso

MMarussi commented on src/screens/LoginScreen/components/WhiteButton/index.js:
```diff
@@ -0,0 +1,15 @@
+
+import styles from './index.module.scss';
+
+function VioletButton({ text }) {
```
El directorio que contiene este componente se llama WhiteButton :thinking:

vivianachoque replied:
Hecho, gracias, buena observación

MMarussi commented on src/screens/LoginScreen/components/FormLogin/index.module.scss:
```diff
@@ -0,0 +1,124 @@
+    flex-direction: column;
+}
+
+.containerbutton {
```
```suggestion
.containerButton {
```

briangerez replied:
Ojo con marcar las conversaciones como resueltas y no resolverlas. Quizás debas subir el código y luego marcarlas mirándolas desde github.

MMarussi reviewed (changes requested):
Bien ahí, lo tenés bastante encaminado.
La mayoría de las sugerencias son detalles, así que no te va a tomar mucho tiempo, vamos!

briangerez commented on src/screens/LoginScreen/components/WhiteButton/index.module.scss:
```diff
@@ -0,0 +1,32 @@
+  background-color: white;
+}
+
+.button:hover {
```
Mismo que .input de más arriba

vivianachoque replied:
Gracias, qué ojo.

briangerez commented on src/screens/LoginScreen/components/FormLogin/index.module.scss:
```diff
@@ -0,0 +1,124 @@
+}
+
+ .error:focus{
+     outline-color: #EA4453;
```
Te recomiendo empezar a usar variables para los colores porque todos los pueden utilizar ya que son globales a toda la página 😄 

Te dejo el link donde se explicó para que lo tengas a mano https://docs.google.com/presentation/d/1f9jJuD61WXPCpIk6kvC56j3qxWf1YbMaFm_M1XxpwAs/edit#slide=id.g6eaa680717_0_126

vivianachoque replied:
Gracias Brian

briangerez commented on .gitignore:
```diff
@@ -1,10 +1,10 @@
 /.pnp
 .pnp.js
-
+s
```
¿Esto está bien?

vivianachoque replied:
corregido

briangerez commented on src/screens/LoginScreen/components/FormLogin/index.js:
```diff
@@ -0,0 +1,92 @@
+
+  handleChange = event => {
+    const { value, name } = event.target;
+    this.setState({ [name]: value });
```
Puedes poner todo en una sola línea:
`this.setState({ [name]: value, errorCode: false, errorName: false, errorServer: false  });`

vivianachoque replied:
Gracias, hecho

briangerez replied:
No lo veo resuelto 👀

briangerez commented on src/screens/LoginScreen/components/FormLogin/index.module.scss:
```diff
@@ -0,0 +1,124 @@
+.formLogin {
+    font-family: 'Montserrat', sans-serif;
```
Esta línea puede ser eliminada porque en el  **src/index.scss** ya se agregó el font-family de toda la app

briangerez reviewed (commented):
Excelente trabajo! 🎉 Te dejo algunos comentarios y sugerencias!

briangerez commented on src/screens/LoginScreen/components/FormLogin/index.module.scss:
```diff
@@ -0,0 +1,124 @@
+ }
+
+.msgCode {
+    font-family: Montserrat;   
```
Creo que puede quitar el font-family de los estilos, ya que es el mismo definido en el index.scss más arriba.

vivianachoque replied:
Resuelto, tenias razon, gracias

vivianachoque committed:
Update src/components/VioletButton/index.js

agregué sugerencia de tincho, cambié disable a disabled

Co-Authored-By: MMarussi <jmmarussi32@gmail.com>

vivianachoque committed:
Update src/screens/LoginScreen/components/FormLogin/index.module.scss

de acuerdo a sugerencia de Tincho, use cammelCase para modificar el nombre de un contenedor

Co-Authored-By: MMarussi <jmmarussi32@gmail.com>

vivianachoque committed:
cambié nombre función violetbutton por WhiteButton ya que el directorio es el de WhiteButton

vivianachoque committed:
Apply suggestions from code review

Tomo sugerencia de Tincho: No es necesario especificar que recibe un evento como parametro ya que es un parámetro implícito.

Co-Authored-By: MMarussi <jmmarussi32@gmail.com>

vivianachoque committed:
modifiqué setState en formLogn/index.js simplifiqué todo en una sola linea

vivianachoque committed:
borré lineas en sass con respecto a font:montserrat ya que por defecto al exportar está configurado

vivianachoque committed:
agregué & a las propiedades de los estilos de cada clase en sus distintos estados

vivianachoque committed:
agregué & a las propiedades de los estilos de cada clase en sus distintos estados

vivianachoque committed:
agregué & a las propiedades de los estilos de cada clase en sus distintos estados

vivianachoque committed:
Corregí código para que quede mas prolijo, uni las clases con sus diferentes estados, pulleé cambios sugeridos por el PR

vivianachoque committed:
Hice cambios en nombre de componente Blockimg a BlockImg

vivianachoque committed:
simplifiqué función handleSubmit

MMarussi commented on src/screens/LoginScreen/components/FormLogin/index.module.scss:
```diff
@@ -0,0 +1,111 @@
+.formLogin {
+    display: flex;
```
En este archivo se usa un identado de cuatro espacio mientras que en otros .scss se usan 2 espacios, no  se cual establecieron como equipo, pero estaría bueno mantener la identación consistente para todos los archivos

vivianachoque committed:
modifiqué padre e hijo en los errores de login ya que estaban mal puestos

vivianachoque committed:
hice cambio en las funciones ya que estaba llamando al evento y era redundante

vivianachoque committed:
modifiqué config de identado, y lo apliqué a todos los archivos de sass

briangerez commented on src/screens/LoginScreen/components/FormLogin/index.module.scss:
```diff
@@ -0,0 +1,111 @@
+}
+
+.title {
+	height: 44px;
```
Otro detalle con la identación. Veo que usan dos espaciados en el resto de los archivos. Deberían mantenerlo en todo el proyecto. Por fa, identar de manera correcta este archivo 😄

vivianachoque committed:
cambie otra vez el tamaño de ident 2 por defecto

vivianachoque committed:
Cambie nombre de componente Blockimg a BlockImg

vivianachoque committed:
cambié otra vez de Blockimg a BlockImg, los cambios no se estarían haciendo

vivianachoque committed:
cambie nombre de Services a services para poder mergear

vivianachoque committed:
cambie nombre de Services a services para poder mergear
